lib/extensions/ar_virtual.rb

```diff
@@ -41,7 +41,7 @@ def arel_attribute(column_name, arel_table = self.arel_table)
 
     def attribute_supported_by_sql?(name)
       load_schema
-      !virtual_attribute?(name) || !!_virtual_arel[name.to_s]
+      attribute_alias?(name) || has_attribute?(name) && (!virtual_attribute?(name) || !!_virtual_arel[name.to_s])
```

Comment:
Doesn't `&&` have precedence over `||` making this line technically `A || (B && (!C || D))`?

Can we clean this up a bit by perhaps moving the last part of this into local variable for readability and also using parens to make it clear what the expected precedence is?